### PR TITLE
Code cleanup 1

### PR DESCRIPTION
**Pull request by gigatexel**
I am currently implementing the EnergyManager2.
When coding, I found that the code could use a cleanup to make it more consistent across classes and improve readability.

- Replaced all "typ" with "type"
- Replaced all `LoxoneEntity.__init__(self, **kwargs)` with `super().__init__(**kwargs)`
- Fixed sensors needing a reboot to start showing values after setting up PyLoxone
- General cleanup of the DeviceInfo attribute. It is now generated by a global function that also monitors if a device already exists. This is particularly usefull for implementations relying on multiple classes (like EnergyManager2)
- Assigning a room/cat is now uniform along classes. In some cases the wrapper function was used, in other cases `add_room_and_cat_to_value_values`
- Entities are now all collected in an `entities` array

What I'm still planning in the next weeks:

- Clean up extra_state_attributes. They are now mostly redundant with the deviceinfo
- See if the .update function can be skipped alltogether when generating entities
- Clean up comments/loggers
- Fix `Entity None (<class 'custom_components.loxone.climate.LoxoneAcControl'>) implements HVACMode(s): off, auto and therefore implicitly supports the turn_on/turn_off methods without setting the proper ClimateEntityFeature. Please create a bug report at https://github.com/JoDehli/PyLoxone/issues`

### CODE REVIEW

**Comment by JoDehli** [comment]
@gigatexel because of all the issues and refactoring I can not work on the new implementation. I is always very time consuming to merge it in the master branch and also in the new implementation branch. I think we should only change really important changes until the new release is out. Maybe at the end of the year if I have time.

**Comment by gigatexel** [comment]
> @gigatexel thanks for your help. But as I already mentioned I would be much better to do all the refactoring if the new version of pyloxone is released. Because I do all the merging work an two different branches.

Is this new-version branch hidden?

**Comment by JoDehli** [comment]
@gigatexel if you want to help me with this I can make it public.

**Comment by gigatexel** [comment]
> @gigatexel if you want to help me with this I can make it public.

Absolutely.
I'm currently working on the implementation of EnergyManager2 and EnergyFlowMonitor. If I can implement then first-time-right in the new version, why not.

**Comment by JoDehli** [comment]
@gigatexel but the integration does not work completely. First the communication must be implemented. The old implementation is not ideal. It was coded quick and dirty from me back in the time. That means new features are not the first thing to do. There must be a lot of work done that it is working again like the old implementation does.

**Comment by gigatexel** [comment]
Ok, I understand. But if you can point to what has to be done/can be done. I have a separate full RPi for debugging etc.
Maybe we should enable the discussions page in this repository :-)

**Comment by JoDehli** [comment]
@gigatexel I created a discussion. I never used it before.

File: custom_components/loxone/lights/colorpickers.py [hunk 1]
```diff
@@ -26,9 +26,9 @@ class RGBColorPicker(LoxoneEntity, LightEntity):
     }
 
     def __init__(self, **kwargs):
-        LoxoneEntity.__init__(self, **kwargs)
+        super().__init__(**kwargs)
         """Initialize the LumiTech."""
-        self._attr_unique_id = self.uuidAction
```
Comment (JoDehli) [inline custom_components/loxone/lights/colorpickers.py:31]: This is not good because we already have a cached variable called unique_id.

**Comment by JoDehli** [comment]
@gigatexel I merged nearly everything manually into the master. Only the self._attr_unique_id I left because I think it is wrong to change it. In the base loxone entity we have a property called  unique_id which gives back the _attr_unique_id. I think this should be like it was before.

**Pull request closed without merging**